extensions/libarchive/BinFiles.cpp

```diff
@@ -259,9 +279,13 @@ void BinFiles::onTrigger(const std::shared_ptr<core::ProcessContext> &context, c
     }
   }
 
-  auto flow = session->get();
+  for (size_t i = 0; i < batchSize_; ++i) {
+    auto flow = session->get();
+
+    if (flow == nullptr) {
+      break;
```

Comment:
the problem with yielding is that, this processor can function correctly even if there are no incoming flowFiles (it can still emit already processed merged files)